### PR TITLE
Superfund -- base import scripts 

### PR DESCRIPTION
**Pull request by sharadshriram**
This PR adds the code used to import the base datasets for EPA superfund sites.
- Location and funding status of different superfund sites on the National Priority List (NPL)
- Hazard and exposure risks to superfund sites from Natural Hazards and the risk to ecology/human health
- Time-series of remedial actions carried-out at different superfund sites based on the contaminated medium

Will update the PR with a more descriptive README file

### CODE REVIEW

**Review by beets — COMMENTED** [review]
Hi Sharad - this is a partial review, but the comments seem to apply for most files so I'll send these out first. Please take a look and let me know if you have any questions.

File: scripts/us_epa/superfund/process_sites.py
Comment (sharadshriram) [reply scripts/us_epa/superfund/process_sites.py]: I have restructured the directory with each sub-directory giving more information on the dataset used for the import, along with steps to download and notes relevant to the import.

File: scripts/us_epa/superfund/process_sites.py
Comment (sharadshriram) [reply scripts/us_epa/superfund/process_sites.py]: Done. I have refactored the block of top-level code to a function which returns a dictionary. With this change, we do away with the global variable `_GEO_MAP`.

File: scripts/us_epa/superfund/process_sites.py
Comment (sharadshriram) [reply scripts/us_epa/superfund/process_sites.py]: The idea is to resolve geoIds for latitude, longitude pairs if and only if a pre-computed map of resolved geoIds for the location is not available. The other reason for this part to be in the except clause is to ensure we speed-up the run-time of the code across multiple runs. This approach worked well during development, but while used by others, the except block is run by default. Hence, while refactoring the top-level code in this script I have removed the exception handling logic.

File: scripts/us_epa/superfund/process_sites.py
Comment (sharadshriram) [reply scripts/us_epa/superfund/process_sites.py]: Fixed the indentation issues in the block comments.

File: scripts/us_epa/superfund/process_sites.py
Comment (sharadshriram) [reply scripts/us_epa/superfund/process_sites.py]: This file was generated in the except block, and the idea for scripting this with exception handling was to ensure that if the map of resolved geoIds is available we use it, which saves a few minutes of running time.

File: scripts/us_epa/superfund/process_sites.py
Comment (sharadshriram) [reply scripts/us_epa/superfund/process_sites.py]: I have made the filename as a constant, now

File: scripts/us_epa/superfund/process_sites.py
Comment (sharadshriram) [reply scripts/us_epa/superfund/process_sites.py]: I have made the filename as a constant, now.

File: scripts/us_epa/superfund/process_sites.py
Comment (sharadshriram) [reply scripts/us_epa/superfund/process_sites.py]: I have updated the code based on this comment.

File: scripts/us_epa/superfund/process_sites.py
Comment (sharadshriram) [reply scripts/us_epa/superfund/process_sites.py]: Did not keep a count here. The duplicates here came from the `site_ownership` dataframe. So we now handle the duplicates at `site_ownership` dataset itself.

File: scripts/us_epa/superfund/process_sites.py
Comment (sharadshriram) [reply scripts/us_epa/superfund/process_sites.py]: Yes, I have the counts. 

We processed 1,715 superfund sites -- initially all of them did not have `establishmentOwnership`. After the merge, we have 1,317 PrivatelyOwned sites, 158 FederalGovernmentOwned sites and 240 sites do not have a value for the establishmentOwnership property.

I have added this note to the script as a comment.

File: scripts/us_epa/superfund/process_sites.py
Comment (sharadshriram) [reply scripts/us_epa/superfund/process_sites.py]: Thanks for this suggestion! I have updated this across all the scripts.

File: scripts/us_epa/superfund/process_sites.py
Comment (sharadshriram) [reply scripts/us_epa/superfund/process_sites.py]: The reason, why I added this check was to run tests without writing to the file and having a the assert check for the expected sites. In hindsight, checking the asserts on the generated csv is better and I have updated the tests. Hence this check is now dropped.

File: scripts/us_epa/superfund/process_sites.py
Comment (sharadshriram) [reply scripts/us_epa/superfund/process_sites.py]: Thanks for pointing this change. I have updated the code.

File: scripts/us_epa/superfund/process_sites_fundingStatus.py
Comment (sharadshriram) [reply scripts/us_epa/superfund/process_sites_fundingStatus.py]: I have updated the directory structure a bit and have added more information about the dataset, how it can be downloaded in the README file in each sub-directory. The README in the directory's root serves as a navigation documentation.

**Comment by sharadshriram** [comment]
@beets , Thank you for the initial review, I have incorporated the comments in the recent commits.

File: .gitignore
Comment (beets) [inline .gitignore:12]: you could also collapse these to:
scripts/us_epa/superfund/**/output

or move this into a .gitignore under scripts/us_epa/superfund

File: scripts/us_epa/superfund/site_and_funding_status/README.md
Comment (beets) [inline scripts/us_epa/superfund/site_and_funding_status/README.md:38]:
```suggestion
1. There is another [EPA Superfund Applet](https://epa.maps.arcgis.com/apps/mapviewer/index.html?layers=c2b7cdff579c41bbba4898400aa38815) that lists more superfund sites but we had difficulties finding a means to export this data to csv.
```

File: scripts/us_epa/superfund/site_contamination/process_sites_contamination.py
Comment (beets) [inline scripts/us_epa/superfund/site_contamination/process_sites_contamination.py:62]:
```suggestion
    # The groupby is handled differently in Python3.7 and Python3.9, hence we have this check. In python3.9, the column is preserved but in 3.7 the column name in groupby is replaced with 0
```

Comment (beets) [reply scripts/us_epa/superfund/site_contamination/process_sites_contamination.py:62]: this is a nice catch -- thanks for documenting.

File: scripts/us_epa/util/superfund_vars.py
Comment (beets) [inline scripts/us_epa/util/superfund_vars.py:1]: please document these constants, and add copyright header.

also, all the entries in this map seem to have key == value. is there a more programmatic way of dealing with this list?

**Review by beets — COMMENTED** [review]
thanks for the updates sharad -- this has come a long way!

File: .gitignore
Comment (sharadshriram) [reply .gitignore:12]: Thank you for the suggestion, I have collapsed the gitignore file.

File: scripts/us_epa/superfund/site_and_funding_status/README.md
Comment (sharadshriram) [reply scripts/us_epa/superfund/site_and_funding_status/README.md:38]: I have added this suggestion to the README

File: scripts/us_epa/superfund/site_contamination/process_sites_contamination.py
Comment (sharadshriram) [reply scripts/us_epa/superfund/site_contamination/process_sites_contamination.py:62]: It was fun getting to this, @ajaits  mentioned that the tests were failing because of the environment and after testing each part of the code mimicking the cloud build environment we learnt of this subtle implementation issue. But I think I will need to rephrase this comment because it is not the groupby method of Python but that of pandas, will update the comment.

File: scripts/us_epa/util/superfund_vars.py
Comment (sharadshriram) [reply scripts/us_epa/util/superfund_vars.py:1]: - added documentation from the constants and the copyright header.

- the elements and compounds we have in the map can be handled programatically since the name corresponds to the dcid after some string manipulation (converting to Title Case). But the `_CHEM_MAP` will start becoming more useful when we add the organic compound contaminants where the name in the dataset will be mapped to a different IUPAC name as dcid. The current implementation is just for us to have all contaminants in one place.

Comment (beets) [reply scripts/us_epa/util/superfund_vars.py:17]: ok, this sounds good.

File: scripts/us_epa/superfund/site_contamination/process_sites_contamination.py
Comment (beets) [reply scripts/us_epa/superfund/site_contamination/process_sites_contamination.py]: ok, please update. i also have a [PR](https://github.com/datacommonsorg/data/pull/613) that i'm about to merge which will update the packages -- please check that the behavior remains as expected. that pr is only possible because of end-to-end tests, which is why we keep asking for them :)

Comment (sharadshriram) [reply scripts/us_epa/superfund/site_contamination/process_sites_contamination.py]: Yes, I have updated the comment in the latest commit [3fb7a665](https://github.com/datacommonsorg/data/pull/586/commits/3fb7a6652e5b7b8e87d502ef25cf20dd87c1d64d), and the cloud build tests pass. I have tested the scripts in this PR locally with Python 3.9 and the upgraded `pandas` version and the tests pass for them too.

**Review by beets — APPROVED** [review]
thanks for the extra testing!

File: scripts/us_epa/superfund/.gitattributes
Comment (beets) [inline scripts/us_epa/superfund/.gitattributes:3]: consider renaming this file?

And perhaps we could add data/*.xlsx , csv, json to lfs

**Review by beets — APPROVED** [review]

File: scripts/us_epa/superfund/.gitattributes
Comment (sharadshriram) [reply scripts/us_epa/superfund/.gitattributes]: Renamed the file.

**Comment by sharadshriram** [comment]
@ajaits , The statvar for contaminants and contaminated thing is now generated from the `csv` file approach which you had mentioned. Can you review this change, please?

File: scripts/us_epa/superfund/site_contamination/process_sites_contamination.py
Comment (beets) [inline scripts/us_epa/superfund/site_contamination/process_sites_contamination.py:133]: you should print out and inspect these errors instead of blindly handling them. there could be other errors...

**Comment by beets** [comment]
Hi Sharad - please avoid adding to a PR once it's already been sent for review (and especially after it's been approved). Small minor fixes for tests are ok, but nothing big should go in which would negate the review.

My expectation, as a review, is that once I approve a review, the PR is merged fairly soon with very very minor updates. Anything new should be done as a follow up PR.

File: scripts/us_epa/superfund/site_contamination/process_sites_contamination.py
Comment (sharadshriram) [reply scripts/us_epa/superfund/site_contamination/process_sites_contamination.py]: Added a print statement to indicate the cause of the exception.

File: scripts/us_epa/superfund/site_contamination/process_sites_contamination.py
Comment (sharadshriram) [reply scripts/us_epa/superfund/site_contamination/process_sites_contamination.py]: Thanks for pointing this out.. I have fixed the formatting of the longer comments

File: scripts/us_epa/superfund/site_contamination/data/test_data/superfund_sites_contamination_expected.csv
Comment (sharadshriram) [reply scripts/us_epa/superfund/site_contamination/data/test_data/superfund_sites_contamination_expected.csv:2]: The use of `&` to concatenate multiple enums as a single value for a property was discussed while importing the Hate Crime statistics. In that import, when `__` was used to concatenate multiple enums/ schema values, there were `dc-import` warnings.

Shall we replace `&` with `__` in a follow-up PR?

**Comment by sharadshriram** [comment]
@beets , Apologies for adding new code after approval, will keep in mind for future PRs to have minimal changes post approval.

File: scripts/us_epa/superfund/site_contamination/data/test_data/superfund_sites_contamination_expected.csv
Comment (beets) [reply scripts/us_epa/superfund/site_contamination/data/test_data/superfund_sites_contamination_expected.csv:2]: yes in a followup. 
what was the warning for `__`? we should add one for `&` too..

**Review by beets — APPROVED** [review]

**Comment by beets** [comment]
> @beets , Apologies for adding new code after approval, will keep in mind for future PRs to have minimal changes post approval.

Preferably no changes after approval :) There might be a few small comments to address - do those then merge right away.

**Pull request merged**
